packages/components/app/styles/components/accordion.scss

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// ACCORDION
+//
+
+@use "../mixins/button" as *;
+@use "../mixins/focus-ring" as *;
+
+:root {
+  --token-accordion-item-padding: 16px;
+  --token-accordion-item-padding-right: 12px;
+}
+
+// ACCORDION COMPONENT (wrapper)
+
+.hds-accordion {
+  display: flex;
+  flex-direction: column;
+  gap: 12px;
+
+  // override default .hds-disclosure-primitive style (Q: Should primitive be refactored?)
+  .hds-accordion-item { width: 100%; }
+}
+
+// ACCORDION ITEM COMPONENT (nested child)
+
+.hds-accordion-item {
+  background: var(--token-color-surface-primary);
+  border: 1px solid;
+  border-radius: 6px;
+
+  // isInteractive=true, Entire AccordionItem is interactive:
+  &.hds-accordion-item--is-interactive {
+    border-color: transparent;
+    box-shadow: var(--token-surface-mid-box-shadow);
+
+    &:hover,
+    &.mock-hover {
+      box-shadow: var(--token-surface-high-box-shadow);
+    }
+  }
+
+  // isInteractive=false, Only nested Chevron button is interactive:
+  &.hds-accordion-item--is--not-interactive {
+    border-color: var(--token-color-border-primary);
+  }
+}
+
+// TOGGLE BLOCK
+
+.hds-accordion-item__toggle {
+  position: relative;
+  display: flex;
+  gap: 12px;
+  align-items: center;
+  padding: var(--token-accordion-item-padding) var(--token-accordion-item-padding) var(--token-accordion-item-padding) var(--token-accordion-item-padding-right);
+}
+
+.hds-accordion-item__button {
+  @include hds-focus-ring-with-pseudo-element();
+  margin: -1px; // offset border size
+  padding: 0;
+
+  &:hover { cursor: pointer; }
+
+  // entire toggle area is interactive
+  &.hds-accordion-item__button--is-not-interactive {
+    position: static;
+    background: transparent;
+    border: 1px solid transparent;
```

Comment:
Yeah, I still see issues with the border making the code much more convoluted. 

@MelSumner do you think would be acceptable to have no borders for this specific button, that has some strict design implementations? (the button is 24px, the icon is 24px, but it has an inner border too, that we can "emulate" but we can't have both at 24px with a 1px transparent border)

Reply:
Adding `margin: -1px;` was the only thing I could come up with. It doesn't seem to cause issues although not ideal...

Reply:
@didoo Do you have any ideas for simplifications/improvements or do you think it is ok to leave as is?

Reply:
I think is OK as it is now. It's clear enough now that we have split the "logic" under two clear class names. Plus, I don't think there's an easy solution that would simplify even more the code. It's the best one can do given the constraints 👍